querydsl-collections/src/main/java/com/querydsl/collections/CollDeleteClause.java

```diff
@@ -33,10 +33,10 @@
 
     private final Path<T> expr;
 
-    private final CollQuery query;
+    private final CollQuery<T> query;
 
     public CollDeleteClause(QueryEngine qe, Path<T> expr, Collection<? extends T> col) {
-        this.query = new CollQuery(qe).from(expr, col);
+        this.query = new CollQuery<Void>(qe).from(expr, col).select(expr);
```

Comment:
Maybe in favor of `new XXXQuery<Void>` we can add static factory methods that return a `XXXQuery<?>`.
Essentially to hide the not yet known parameter.
I recall having discussed this, but I think it's good to recap.